windows-gaming-driver/src/driver/controller.rs

```diff
@@ -219,6 +223,24 @@ impl Controller {
         }
     }
 
+    pub fn temporary_entry(&mut self, sender: UnboundedSender<(i32, i32)>, x: i32, y: i32) -> bool {
+        if self.temporary_entry_writer.is_none() {
+            self.temporary_entry_writer = Some(sender);
+            self.write_ga(GaCmd::SetMousePosition(x, y));
+            self.light_attach();
+            return true;
+        }
+        false
+    }
+
+    pub fn temporary_exit(&mut self) {
+        // only detach if we are not already detached
+        if self.temporary_entry_writer.is_some() {
+            self.io_detach();
```

Comment:
What exactly do you thik this could race with? The `UnboundedReceiver` [will first return all buffered messages and then return `None`](https://alexcrichton.com/futures-rs/src/futures/unsync/mpsc.rs.html#179).